### PR TITLE
Integrate wild carrots, groundnuts and jerusalem artichokes into more recipes

### PR DESCRIPTION
**Pull request by NetSysFire**
#### Summary
None

#### Purpose of change

Some more optimizations to the recipes and a couple of basic new ones.

#### Describe the solution

I did some research on recipes.

- It appears you can glaze carrots with both oil and butter. Since this is the apocalypse, the little bit of difference in taste is irrelevant.
- Original source for the fruit mash: https://www.topinambur-manufaktur.de/desserts
- I originally also added a bunch of groundnut recipes but @pjf did #60067 before I could submit this. Source for the groundnut crisps: https://www.youtube.com/watch?v=MB3wW_VE9yg
- Unfortunately I am not able to find that link again but I am fairly sure this was a blog post about how you could add groundnut into your scrambled eggs, just like potatoes.
- Also adds some comestible types I forgot last time.

#### Testing

Loads fine locally.

### CODE REVIEW

File: data/json/recipes/recipe_food.json [hunk 1]
```diff
@@ -4359,7 +4383,14 @@
...
       [ [ "eggs_bird", 2, "LIST" ], [ "egg_reptile", 2 ] ],
       [ [ "any_butter_or_oil", 2, "LIST" ] ],
-      [ [ "mushroom", 1 ], [ "mushroom_morel", 1 ], [ "garlic_clove", 1 ], [ "onion", 1 ] ],
+      [
+        [ "mushroom", 1 ],
+        [ "mushroom_morel", 1 ],
+        [ "garlic_clove", 1 ],
+        [ "wild_garlic", 1 ],
+        [ "onion", 1 ],
+        [ "groundnut", 1 ]
```
Comment (pjf) [inline data/json/recipes/recipe_food.json:4392]: I'd consider using either `groundnut_prepared` or `veggy_potatolike` here.

The first requires the ground nuts be peeled and lightly boiled (to destroy protease inhibitors), the second does the same but also allows real potatoes to be used.

```suggestion
        [ "veggy_potatolike", 1, "LIST" ]
```

**Review by pjf — CHANGES_REQUESTED** [review]

**Review by Maleclypse — APPROVED** [review]

File: data/json/items/comestibles/raw_veggy.json [hunk 2]
```diff
@@ -82,6 +82,7 @@
     "type": "COMESTIBLE",
     "id": "jerusalem_artichoke",
     "name": { "str": "Jerusalem artichoke" },
+    "comestible_type": "FOOD",
```
Comment (Fris0uman) [inline data/json/items/comestibles/raw_veggy.json:85]:
```suggestion
```

File: data/json/items/comestibles/raw_veggy.json [hunk 3]
```diff
@@ -91,7 +92,6 @@
     "calories": 33,
     "spoils_in": "20 days",
     "description": "A slightly sweet tuber.  Contains much inulin which may cause some minor issues with your digestive tract, so it should be cooked.",
-    "comestible_type": "FOOD",
```
Comment (Fris0uman) [inline data/json/items/comestibles/raw_veggy.json:94]:
```suggestion
    "comestible_type": "FOOD",
```
Moving this around is unecessary

File: data/json/items/comestibles/raw_veggy.json [hunk 4]
```diff
@@ -72,7 +73,6 @@
     "stim": 1,
     "spoils_in": "20 days",
     "description": "This aromatic rhizome can be eaten as is in an emergency, but it is much better used for making beverages.  Its taste is pleasant but it is also tough and stringy.",
-    "comestible_type": "FOOD",
```
Comment (Fris0uman) [inline data/json/items/comestibles/raw_veggy.json:75]:
```suggestion
    "comestible_type": "FOOD",
```
Moving this around is unecessary

File: data/json/items/comestibles/raw_veggy.json [hunk 5]
```diff
@@ -62,7 +62,8 @@
     "type": "COMESTIBLE",
     "id": "wild_sarsaparilla_root",
     "name": { "str": "wild sarsaparilla root" },
-    "weight": "50 g",
+    "comestible_type": "FOOD",
```
Comment (Fris0uman) [inline data/json/items/comestibles/raw_veggy.json:65]:
```suggestion
```

File: data/json/items/comestibles/raw_veggy.json (hunk 2 above)
Comment (NetSysFire) [reply data/json/items/comestibles/raw_veggy.json:85]: Ugh, looks like the result of the rebase. Good catch.

**Comment by Maleclypse** [comment]
You've failed calorie and density tests on jerusalem artichokes and maybe something else.

**Comment by NetSysFire** [comment]
I saw the failed tests but real life got in my way. I already took a shot at fixing that but failed, mainly being confused about what calories they should be and considering just adding NUTRIENT_OVERRIDE.

**Comment by Maleclypse** [comment]
No stress. I’ll work on fixing the tests. The calorie ones are particularly difficult sometimes.

**Comment by NetSysFire** [comment]
@PatrikLundell can you look over the latest commit (re https://github.com/CleverRaven/Cataclysm-DDA/issues/60599#issuecomment-1232681973)?

I removed cholla buds from stuff because even though its a fruit it is not in any of the component groups, making this part effectively unreachable. I have my doubt on its though, seeing that it will get inherited by mods but honestly it does not look right to me to add every mod fruit into `conditional_names` because it gets cluttered quickly.

**Comment by PatrikLundell** [comment]
What's added looks reasonable (assuming those thingies can be processed these ways). When it comes to adding stuff in various places, I'm leaning towards completeness, rather than leaving people guessing whether something was left out on purpose or whether it was unintentional (especially without proper comment support...).
The difficult part in reviewing is to find the things that aren't there but should be, but I failed to spot any of those.

Am I right in assuming that's the beginning (with veggies, if any, and drinks to follow)?

**Comment by NetSysFire** [comment]
>(assuming those thingies can be processed these ways)

They are in `sweet_fruit_fresh`, which is in `sweet_fruit` (`requirements/cooking_components.json`), which is what all of the recipes are using.
If a recipe is just using one of those blanket groups, you can basically copy paste.

>Am I right in assuming that's the beginning (with veggies, if any, and drinks to follow)?

Quite possibly. Checking all of the lists is going to be super exhausting and tedious, and I wonder if it wouldnt be faster to just write a test for it. However I already have some problematic cases in my head: There is no way for the test to know which component list is to be used (e.g sugar/water vs all the fruits) or what recipes need conditional names.

**Comment by PatrikLundell** [comment]
Yes, it is tedious, and it doesn't help that no files have their entries in the same order.

**Comment by Maleclypse** [comment]
OK so I'm looking at your calorie issue and I think your issue is that jerusalem artichokes raw, you are saying they have less available calories than the cooked version?   If that's the case you need to use `NUTRIENT_OVERRIDE` otherwise the calorie calculations are trying to use the raw calorie value to calculate the total.  Unfortunately that means that multiple ingredient recipes that include raw jerusalem artichokes will also need `NUTRIENT_OVERRIDE` as well unless they use the cooked result as an ingredient instead of the raw version.  Does that make sense?

**Comment by NetSysFire** [comment]
First of all, thanks for looking a this and taking the time to debug.

>you are saying they have less available calories than the cooked version?

Could this be caused by my `copy-from` shenanigans and then deleting the RAW flag? Does the calorie value of the raw jerusalem artichoke have to include the 25% penalty from RAW already? Let us pretend a raw tuber would be 100 kcal for simplicity: does it need to be 125 kcal because to make up for RAW?

>Does that make sense?

Not yet. As far as I know the test derives the calories from the ingredients. So e.g boiled jerusalem artichoke (100 kcal) plus blueberries (89 kcal) in the fruitmash would equal 189 kcal. I would go for 200 kcal here to be generous (also cooking magic™), maybe even a bit more so it actually makes sense crafting it instead of just eating the ingredients as is.

I know I previously went for 150 and then straight 250 because I had no idea and was randomly trying stuff. Now I would go for 200 after I solve the conflicts which will not be today because my irl weariness is high. The density test should be easy enough to fix since I just need to multiply the volume by 1.1.

**Comment by NetSysFire** [comment]
@PatrikLundell I reverted my changes to the file because it appears someone already did it in master /o\
The good news is that I could resolve that scary merge conflict easily.

**Comment by NetSysFire** [comment]
>Average recipe Calories must fall within this range, derived from default Calories: 600-1000

I really do not understand get where it gets this value. Maybe I should really set `NUTRIENT_OVERRIDE` to the fruitmash but no idea about the boiled and sauteed variant.

**Comment by I-am-Erk** [comment]
Nutrition tests are rough. Can you bring this up on discord where we can look in realtime? Otherwise we may need to freeze this one.

**Comment by NetSysFire** [comment]
>Can you bring this up on discord where we can look in realtime?

I dont have discord. But I am on IRC.

**Comment by Wishbringer** [comment]
> > Can you bring this up on discord where we can look in realtime?
> 
> I dont have discord. But I am on IRC.

Discord has a web interface, no need to install anything.

**Comment by Maleclypse** [comment]
The sautéed artichoke either remove the two charges or set calorie override. The density check is copying from an item with a known bad dendity

**Comment by NetSysFire** [comment]
>Discord has a web interface, no need to install anything.

Not interested in offering my phone number to this service.

>The sautéed artichoke either remove the two charges or set calorie override. 

Thanks for the hint! Just pushed.

**Comment by Maleclypse** [comment]
Sorry I wasn’t clear. Two test failures. One was a calorie test the charges was intended to fix. The density fix I can’t solve from my phone. Sorry

**Comment by Maleclypse** [comment]
Good news the charges fixed the sautéed error. The boiled artichoke you probably want to make a nutrient override flag or change calories to 30. But I think you want the flag.

**Comment by NetSysFire** [comment]
Thanks for the help, I just added the flag and hope it will not break recipes or something.

**Comment by Zireael07** [comment]
Offtop: @NetSysFire I never had to give Discord my mobile number, not even when installing on said mobile.

**Comment by NetSysFire** [comment]
I am most certainly heavily biased, but last time I looked it required a phone number. I am very anti-discord (blah blah proprietary and privacy), although I wouldnt mind a discord bridge to the dev channel in the IRC, since this one does not require registration to chat, except for the bot of course.

**Pull request merged**